### PR TITLE
moving transformusername to common

### PR DESCRIPTION
**Pull request by tinakurian**
Moving GenerateCompliantUsername and TransofmrUsername functions to common. GenerateCompliantUsername can then be used in registration-service when checking for crtadmin users.

This work is related to CRT-913: preventing usersignups for usernames that have the crtadmin suffix.

Related PRs: codeready-toolchain/toolchain-e2e#237, codeready-toolchain/toolchain-common#151,
codeready-toolchain/registration-service#170

Jira: https://issues.redhat.com/browse/CRT-913

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -33,6 +33,7 @@ require (
 
 replace (
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v13.3.2+incompatible // Required by OLM
+	github.com/codeready-toolchain/toolchain-common v0.0.0-20210209125948-7a9e8c9e4ee6 => github.com/tinakurian/toolchain-common v0.0.0-20210216182308-9a7445b1611e
```
Comment (alexeykazakov) [inline go.mod:36]: Reminder.

**Review by alexeykazakov — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -33,6 +33,7 @@ require (
 
 replace (
 	github.com/Azure/go-autorest => github.com/Azure/go-autorest v13.3.2+incompatible // Required by OLM
+	github.com/codeready-toolchain/toolchain-common v0.0.0-20210209125948-7a9e8c9e4ee6 => github.com/tinakurian/toolchain-common v0.0.0-20210217164217-8325ca100257
```
Comment (xcoulon) [inline go.mod:36]: reminder ;)

File: pkg/controller/usersignup/usersignup_controller.go [hunk 3]
```diff
@@ -5,7 +5,7 @@ import (
 	"crypto/md5"
 	"encoding/hex"
 	"fmt"
-	"regexp"
+	"github.com/codeready-toolchain/toolchain-common/pkg/usersignup"
```
Comment (xcoulon) [inline pkg/controller/usersignup/usersignup_controller.go:8]: could you move this import declaration to line 11, with all the other `github.com/codeready-toolchain/...` ones?

**Comment by xcoulon** [comment]
looks good overall, just a minor comment and a reminder

File: pkg/controller/usersignup/usersignup_controller.go (hunk 3 above)
Comment (tinakurian) [reply pkg/controller/usersignup/usersignup_controller.go]: fixed here: 16c3c1b7a8c33fe7504c3f58a757a168b7acef05

Comment (xcoulon) [reply pkg/controller/usersignup/usersignup_controller.go]: thanks ;)

**Review by xcoulon — APPROVED** [review]

**Pull request merged**